### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by sind14**

### CODE REVIEW

File: app/Knight_stat/calc_knight_stat.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    power += knight["power"] + knight["weapon"]["power"]
+    for armour in knight["armour"]:
+        hp += armour["protection"]
+    if knight["potion"] is not None:
+        if "power" in knight["potion"]["effect"]:
+            power += knight["potion"]["effect"]["power"]
+        if "protection" in knight["potion"]["effect"]:
+            hp += knight["potion"]["effect"]["protection"]
+        if "hp" in knight["potion"]["effect"]:
+            hp += knight["potion"]["effect"]["hp"]
```
Comment (ArsenPidhoretskyi) [inline app/Knight_stat/calc_knight_stat.py:20]: Use for loop instead of duplicate code

File: app/main.py [hunk 2]
```diff
@@ -1,213 +1,29 @@
...
-    # check if someone fell in battle
-    if arthur["hp"] <= 0:
-        arthur["hp"] = 0
+def battle(solution: set[tuple]):
+    lancelot_hp_after = Lancelot_stat[0] - Mordred_stat[1]
+    mordred_hp_after = Mordred_stat[0] - Lancelot_stat[1]
+    if lancelot_hp_after <= 0:
+        lancelot_hp_after += abs(lancelot_hp_after)
+    if mordred_hp_after <= 0:
+        mordred_hp_after += abs(mordred_hp_after)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:13]: Create additional function for fighting with two knights

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]
Others questions was answered in chat

File: app/Knight_stat/calc_knight_stat.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    power += knight["power"] + knight["weapon"]["power"]
+    for armour in knight["armour"]:
+        hp += armour["protection"]
+    if knight["potion"] is not None:
+        for potion in knight["potion"]["effect"]:
+            if potion == "power":
+                power += knight["potion"]["effect"]["power"]
+            else:
+                hp += knight["potion"]["effect"][potion]
+    return (hp, power)
```
Comment (Bohdan-Salamakha) [inline app/Knight_stat/calc_knight_stat.py:14]: It's better to return not tuple but dict

File: app/Knight_stat/one_to_one.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+def one_to_one(knights: list[tuple]) -> list[int]:
+    lancelot_hp_after = knights[0][0] - knights[2][1]
+    mordred_hp_after = knights[2][0] - knights[0][1]
+    if lancelot_hp_after <= 0:
+        lancelot_hp_after += abs(lancelot_hp_after)
+    if mordred_hp_after <= 0:
+        mordred_hp_after += abs(mordred_hp_after)
+    arthur_hp_after = knights[1][0] - knights[3][1]
+    red_knight_hp_after = knights[3][0] - knights[1][1]
```
Comment (Bohdan-Salamakha) [inline app/Knight_stat/one_to_one.py:9]: It's better to create separated variables but not use double indexation

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**